presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java

```diff
@@ -253,25 +227,23 @@ public static NullableValue getColumnValueForPartitionKey(Row row, int position,
             case TIMEUUID:
                 return NullableValue.of(nativeType, utf8Slice(row.getUUID(position).toString()));
             default:
-                return getColumnValue(row, position, cassandraType, typeArguments);
+                return getColumnValue(row, position, cassandraType);
         }
     }
 
-    private static String buildSetValue(Row row, int position, CassandraType elemType)
+    private static String buildMapValue(Row row, int position)
     {
-        return buildArrayValue(row.getSet(position, elemType.javaType), elemType);
+        Definition definition = row.getColumnDefinitions().asList().get(position);
```

Comment:
You can replace `.asList().get(position).getType()` with `.getType(position)`

```suggestion
        Type type = row.getColumnDefinitions().getType(position);
```